arrays/arrays.c

```diff
@@ -99,15 +99,25 @@ char *arr_read(Array *arr, int index) {
 void arr_insert(Array *arr, char *element, int index) {
 
   // Throw an error if the index is greater than the current count
-
+  if(index >= arr->count){
+    printf("Array capacity has been reached");
+    return 0;
+  }
   // Resize the array if the number of elements is over capacity
-
+  if(arr->count+1 > arr->capacity){
+    resize_array(arr);
+  }
+  
   // Move every element after the insert index to the right one position
-
+  for(int i = index +1; i<arr->count; i++){
```

Comment:
This part here needs to be corrected. Try using python tutor to see exactly what is happening during this step of the arr_insert function.